test-utils/state-viewer/src/state_dump.rs

```diff
@@ -0,0 +1,201 @@
+use near_chain::RuntimeAdapter;
+use near_chain_configs::{Genesis, GenesisConfig};
+use near_primitives::block::BlockHeader;
+use near_primitives::state_record::StateRecord;
+use near_primitives::types::{AccountInfo, StateRoot};
+use near_store::TrieIterator;
+use neard::NightshadeRuntime;
+use std::collections::HashMap;
+
+pub fn state_dump(
+    runtime: NightshadeRuntime,
+    state_roots: Vec<StateRoot>,
+    last_block_header: BlockHeader,
+    genesis_config: &GenesisConfig,
+) -> Genesis {
+    println!("Generating genesis from state data");
+    let genesis_height = last_block_header.inner_lite.height + 1;
+    let block_producers = runtime
+        .get_epoch_block_producers_ordered(
+            &last_block_header.inner_lite.epoch_id,
+            &last_block_header.hash,
+        )
+        .unwrap();
+    let validators = block_producers
+        .into_iter()
+        .filter_map(|(info, is_slashed)| {
+            if !is_slashed {
+                Some((info.account_id, (info.public_key, info.stake)))
+            } else {
+                None
+            }
+        })
+        .collect::<HashMap<_, _>>();
+
+    let mut records = vec![];
+    for state_root in &state_roots {
+        let trie = TrieIterator::new(&runtime.trie, &state_root).unwrap();
+        for item in trie {
+            let (key, value) = item.unwrap();
+            if let Some(mut sr) = StateRecord::from_raw_key_value(key, value) {
+                if let StateRecord::Account { account_id, account } = &mut sr {
+                    if account.locked > 0 {
+                        if let Some((_, stake)) = validators.get(account_id) {
```

Comment:
can be just `let stake = get(account_id).or_else(&0)`, because `amount = amount + locked - stake` works in both cases.